pylxd/client.py

```diff
@@ -190,9 +191,19 @@ class Client(object):
     def __init__(self, endpoint=None, version='1.0', cert=None, verify=True):
         self.cert = cert
         if endpoint is not None:
-            if endpoint.startswith('/') and os.path.isfile(endpoint):
+            if (
+                endpoint.startswith('/') and (
+                    os.path.isfile(endpoint) or (
+                        os.path.exists(endpoint) and
+                        stat.S_ISSOCK(os.stat(endpoint).st_mode)
+                    )
+                )
+            ):
                 self.api = _APINode('http+unix://{}'.format(
                     parse.quote(endpoint, safe='')))
+            elif endpoint.startswith('http+unix://') and '/' in endpoint[12:]:
```

Comment:
I would rather not expose the `http+unix://` part of the url to api consumers. Recently, I've had my faith in requests-unixsocket shaken, and there are other libraries that do similar things with different protocol names. Switching dependencies would become harder if we exposed `http+unix://` as a supported option.